src/Identity/Extensions.Core/src/UserManager.cs

```diff
@@ -176,6 +177,11 @@ public UserManager(IUserStore<TUser> store,
     /// </summary>
     public IdentityOptions Options { get; set; }
 
+    /// <summary>
+    /// The <see cref="IServiceProvider"/> used to resolve Identity services.
+    /// </summary>
+    public IServiceProvider ServiceProvider { get; }
```

Comment:
Did this get an API review? It's unfortunate that SignInManger isn't in the same assembly so we cannot just share it internally. I think this is probably better than the alternatives of adding a new SignInManager constructor or using SignInManager.Context.RequestServices, but it might be worth discussing with a wider audience to get more ideas. I don't think we need to block the PR on this though since we can make follow up changes.